contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -638,6 +640,14 @@ init_tsql_coerce_hash_tab(PG_FUNCTION_ARGS)
 
 						if (!OidIsValid(entry->castfunc))
 						{
+						   /*
+							* varbinary to binary implicit type cast without function should be allowed
+							* during upgrade since the cast function might not exists in older version
+							*/ 
+							if (babelfish_dump_restore && ((*common_utility_plugin_ptr->is_tsql_varbinary_datatype) (castsource) 
+								&& (*common_utility_plugin_ptr->is_tsql_binary_datatype) (casttarget)))
+								entry->castfunc = 0;
+
```

Comment:
can you describe use case and explain why this changes are required? And what error or differences you will see if we do not make this change?